src/ctap/data_formats.rs

```diff
@@ -281,23 +281,23 @@ impl Extensions {
     pub fn has_make_credential_hmac_secret(&self) -> Result<bool, Ctap2StatusCode> {
         self.0
             .get("hmac-secret")
-            .map(read_bool)
+            .map(|b| extract_bool(b.clone()))
             .unwrap_or(Ok(false))
     }
 
     pub fn get_assertion_hmac_secret(
-        &self,
+        mut self,
```

Comment:
This is a semantics change. If we really expect to call this function exactly once per `Extensions` we should rename this function to `extract_` instead of `get_`. Otherwise we should simply clone the output: `self.0.get("hmac").map(|x| try_from(x.clone())`.

Same for the function below.